src/main/java/edu/harvard/iq/dataverse/util/SignpostingResources.java

```diff
@@ -0,0 +1,337 @@
+        }
+
+        if (!Objects.equals(ds.getPersistentURL(), "")) {
+            String citeAs = "<" + ds.getPersistentURL() + ">;rel=\"cite-as\"";
+            valueList.add(citeAs);
+        }
+
+        List<FileMetadata> fms = workingDatasetVersion.getFileMetadatas();
+        String items = getItems(fms);
+        if (items != null && !Objects.equals(items, "")) {
+            valueList.add(items);
+        }
+
+        String describedby = "<" + describedByJsonObj.getString(ds.getProtocol()) + ds.getAuthority() + "/"
+                + ds.getIdentifier() + ">;rel=\"describedby\"" + ";type=\"" + describedByJsonObj.getString("type") + "\"";
+        describedby += ",<" + systemConfig.getDataverseSiteUrl() + "/api/datasets/export?exporter=schema.org&persistentId="
+                + ds.getProtocol() + ":" + ds.getAuthority() + "/" + ds.getIdentifier() + ">;rel=\"describedby\"" + ";type=\"application/json+ld\"";
+        valueList.add(describedby);
+
+        String type = "<https://schema.org/AboutPage>;rel=\"type\"";;
+        if (useDefaultFileType) {
+            type = "<https://schema.org/AboutPage>;rel=\"type\",<" + defaultFileTypeValue + ">;rel=\"type\"";
+        }
+        valueList.add(type);
+
+        // TODO: support only CC0 now, should add flexible license support when flex-terms or multi-license is ready
+//        License license = workingDatasetVersion.getTermsOfUseAndAccess().getLicense();
+        License license = workingDatasetVersion.getTermsOfUseAndAccess().getLicense();
+        String licenseString = license.getUri() + ";rel=\"license\"";
+        valueList.add(licenseString);
+
+        /*
+        else {
+            TODO: when multilicense is merged, should get license link from multilicense
+             Thus the checking thus would not be necessary?
+             Currently, when the License is None, which means bring-your-own-license,
+             Mostly, we will not have a valid URI in the License, so we skip it here despite that signposting requires 1
+             valid URI.
+        }
+        */
+
+        String linkset = "<" + systemConfig.getDataverseSiteUrl() + "/api/datasets/:persistentId/versions/"
+                + workingDatasetVersion.getVersionNumber() + "." + workingDatasetVersion.getMinorVersionNumber()
+                + "/linkset?persistentId=" + ds.getProtocol() + ":" + ds.getAuthority() + "/" + ds.getIdentifier() + "> ; rel=\"linkset\";type=\"application/linkset+json\"";
+        valueList.add(linkset);
+        logger.info(String.format("valueList is: %s", valueList));
+
+        return String.join(", ", valueList);
+    }
+
+    private String getAuthorUrl(DatasetAuthor da) {
+        String authorURL = "";
+        if (da.getIdValue() != null && !da.getIdValue().trim().isEmpty()) {
+            authorURL = da.getIdValue();
+        } else if (da.getIdentifierAsUrl() != null && !da.getIdentifierAsUrl().trim().isEmpty()) {
+            authorURL = da.getIdentifierAsUrl();
+        }
```

Comment:
Looks to me like getIdentifierAsUrl already includes this type of checking (could be new since when you started coding) - is this still needed?